x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/IndexResolver.java

```diff
@@ -125,22 +129,64 @@ public void resolveAsMergedMappingAndRetrieveMinimumVersion(
             EsqlResolveFieldsAction.TYPE,
             createFieldCapsRequest(indexWildcard, fieldNames, requestFilter, includeAllDimensions),
             listener.delegateFailureAndWrap((l, response) -> {
-                TransportVersion minimumVersion = response.minTransportVersion();
-
-                LOGGER.debug("minimum transport version {}", minimumVersion);
-                l.onResponse(
-                    new Versioned<>(
-                        mergedMappings(indexWildcard, new FieldsInfo(response.caps(), supportsAggregateMetricDouble, supportsDenseVector)),
-                        // The minimum transport version was added to the field caps response in 9.2.1; in clusters with older nodes,
-                        // we don't have that information and need to assume the oldest supported version.
-                        minimumVersion == null ? TransportVersion.minimumCompatible() : minimumVersion
-                    )
+                FieldsInfo info = new FieldsInfo(
+                    response.caps(),
+                    response.minTransportVersion(),
+                    Build.current().isSnapshot(),
+                    supportsAggregateMetricDouble,
+                    supportsDenseVector
                 );
+                LOGGER.debug("minimum transport version {} {}", response.minTransportVersion(), info.effectiveMinTransportVersion());
+                l.onResponse(new Versioned<>(mergedMappings(indexWildcard, info), info.effectiveMinTransportVersion()));
             })
         );
     }
 
-    public record FieldsInfo(FieldCapabilitiesResponse caps, boolean supportAggregateMetricDouble, boolean supportDenseVector) {}
+    /**
+     * Information for resolving a field.
+     * @param caps {@link FieldCapabilitiesResponse} from all indices involved in the query
+     * @param minTransportVersion the minimum {@link TransportVersion} of any node that <strong>might</strong> receive the request
```

Comment:
More precisely: isn't this the minimum transport version of the respective *cluster* for each node that might be a data node for this request?

There are more nodes participating (ml nodes, lookup nodes), and this piece of info is important to be able to reason about their versions.

Reply:
I'll update the comment. It's the minimum transport version of all nodes in all target clusters. So if there are ml nodes that are an older version this will come back with their version.